src/Exception/UnencodableValueException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace MongoDB\Exception;
+
+use InvalidArgumentException;
+
+use function get_debug_type;
+use function sprintf;
+
+class UnencodableValueException extends InvalidArgumentException implements Exception
+{
+    /** @var mixed */
+    private $value;
+
+    /** @param mixed $value */
+    public function __construct($value)
```

Comment:
_If_ we end up sharing exceptions for `encode` and `decode`, I'd suggest making the constructor private and creating factory methods like we do for other exceptions.

I might suggest doing this regardless, as it keeps this non-internal class more flexible. By overriding the constructor here, you're making it difficult to ever pass original constructor arguments to InvalidArgumentException (e.g. code, previous exception).

My suggested factory method names would be:

 * If a common UnsupportedValueException: `invalidEncodableType()` and `invalidDencodableType()`
 * If separate classes: `invalidType()`